router/router.go

```diff
@@ -19,40 +17,58 @@ const (
 
 type (
 	router struct {
-		mux     *http.ServeMux
-		gateway gateway
-		config  config.RouterConfig
-		logger  polylog.Logger
+		mux           *http.ServeMux
+		gateway       gateway
+		healthChecker *healthChecker
+
+		config config.RouterConfig
+
+		logger polylog.Logger
 	}
 	gateway interface {
 		HandleHTTPServiceRequest(ctx context.Context, httpReq *http.Request, w http.ResponseWriter)
 	}
 )
 
+type RouterParams struct {
```

Comment:
Challenging this. Is it really necessary?

I think passing all of it to the factory function is easier one by one.

Reply:
Sure. I've updated.